beta-src/src/components/ui/WDMessageList.tsx

```diff
@@ -24,27 +21,50 @@ const WDMessageList: React.FC<WDMessageListProps> = function ({
   countryIDSelected,
   messagesEndRef,
 }): React.ReactElement {
-  const [viewport] = useViewport();
-  const device = getDevice(viewport);
-  const height = "350px";
   const filteredMessages = messages.filter(
     (message) =>
       (message.fromCountryID === countryIDSelected ||
         message.toCountryID === countryIDSelected) &&
       (countryIDSelected === 0 || message.toCountryID !== 0), // public messages in public chat
   );
-  const messageComponents = filteredMessages.map((message: GameMessage) => (
-    <WDMessage
-      key={`${message.timeSent}:${message.fromCountryID}:${message.toCountryID}:${message.message}`}
-      message={message}
-      userCountry={userCountry}
-      allCountries={allCountries}
-    />
-  ));
+  const messagesByTurn: { [key: number]: React.ReactElement[] } = {};
+  filteredMessages.forEach((message: GameMessage) => {
+    if (!(message.turn in messagesByTurn)) {
+      messagesByTurn[message.turn] = [];
+    }
+    messagesByTurn[message.turn].push(
+      <WDMessage
+        key={`${message.timeSent}:${message.fromCountryID}:${message.toCountryID}:${message.message}`}
+        message={message}
+        userCountry={userCountry}
+        allCountries={allCountries}
+      />,
+    );
+  });
+  const messageTurnComponents = Object.entries(messagesByTurn).map(
```

Comment:
You need a key attribute here now because you're adding a list of boxes to the rendering.